liblouis/commonTranslationFunctions.c

```diff
@@ -43,22 +43,26 @@ _lou_handlePassVariableTest(const widechar *instructions, int *IC, int *itsTrue)
 		return 1;
 
 	case pass_lt:
-		if (passVariables[instructions[*IC + 1]] >= instructions[*IC + 2]) *itsTrue = 0;
+		if ((widechar)passVariables[instructions[*IC + 1]] >= instructions[*IC + 2])
```

Comment:
Can you just cast an `int` into a `widechar` aka a `unsigned short int` (with ucs2) or a `unsigned int` (with ucs4)? Isn't that asking for trouble?

Reply:
AH yes, good catch. It shouldn't be problematic for ucs-4 because passVariables items always seems positive, but with unsigned short int, we have only 16 bits.

Given the code, it would make sense to make passVariables either unsigned int or widechar, which was my initial try. But, that seems to break tests.

Would you agree with casting to unsigned int here?

Reply:
TBH, I don't quite understand what this code is doing, but from looking at it I would assume that `passVariables` should be of type `widechar`

Reply:
I see now that the tests `multipass-backward` and `multipass-forward` fail when I use `static widechar passVariables[NUMVAR];`. @bertfrees could that be a real problem in the multipass code?

Reply:
The way I understand it, `passVariables` is an array of variable values (the variable keys are the array indices). The type was chosen to be `int`, but could as well have been an `unsigned short` or `unsigned byte` because the values are typically small. UCS-2 or UCS-4 has nothing to do with it because it is never characters that are assigned to variables. After compilation, the variable keys and values from the rules just happen to be stored in the `passInstructions` array which is of type `widechar`, however `passInstructions` is not only used to store characters and dot patterns. It contains the whole compiled multipass rule. The variable keys come from the `passGetVariableNumber` function which parses a part of a multipass rule into a number between 0 and `NUMVAR` (50). The initial variable values come from the `passGetNumber` function which parses a part of a multipass rule into a non-negative number (which could in theory overflow an `int` but in practice the values will be small). During a translation, variable values can also be incremented or decremented.

In other words, I think the line above should become:

```c
if (passVariables[instructions[*IC + 1]] >= (int)instructions[*IC + 2])
```

Reply:
Thanks for the clarification. My first reaction was to change `passInstructions` to `int` instead of `widechar`. However, as it sounds like `passInstructions` is some kind of multi-purpose data structure that can also hold `widechar`, we probably shouldn't do that. Probably should have been some union datatype that reflects this intention. 

But anyway, the `int` cast looks like the way to go